### ISSUE TITLE
latest Jinja2 (2.7) requires Python 2.6 or later

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#1173]
Currently, the requirements in setup.py simply demand  "Jinja2>=2.3"; however, this breaks Python 2.4, 2.5, 3.1, and 3.2, which are not supported by Jinja2 2.7:

http://jinja.pocoo.org/docs/changelog/#version-2-7

Ideally, for these older Python versions, the requirement could be stated instead as 'Jinja2>=2.3,<2.7'  I'm not sure if the requirement in setup.py can be simply made conditional on the Python version with something like this:

if sys.hexversion < 0x02060000 or 0x03000000 <= sys.hexversion < 0x03030000:
    requires = ['Pygments>=1.2', 'Jinja2>=2.3,<2.7', 'docutils>=0.7']
else:
    requires = ['Pygments>=1.2', 'Jinja2>=2.3', 'docutils>=0.7']

but that might be worth a try.

It might also be worth adding some kind of mention of this issue to http://sphinx-doc.org/intro.html#prerequisites

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/1173
- Originally reported by: Alexander Dupuy
- Originally created at: 2013-05-21T13:27:22.868

**Comment by shimizukawa** [comment]
_From Alexander Dupuy on 2013-05-21 11:33:31+00:00_

Sorry, omitted Markdown code-block for example code above, here it is a bit more readably:

```
#!python

if sys.hexversion < 0x02060000 or 0x03000000 <= sys.hexversion < 0x03030000:
    requires = ['Pygments>=1.2', 'Jinja2>=2.3,<2.7', 'docutils>=0.7']
else:
    requires = ['Pygments>=1.2', 'Jinja2>=2.3', 'docutils>=0.7']
```

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2013-05-29 09:17:05+00:00_

Fix Python version compatibility with Jinja2 2.7. closes #1173

→ <<cset b2997a652ea2c8a756f31e34263fc6d6de8eaade>>

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2013-05-29 09:17:49+00:00_

[Alexander Dupuy](https://bitbucket.org/dupuy) Thanks!
